csharp/src/ProtocolBuffers.Test/Collections/RepeatedFieldTest.cs

```diff
@@ -239,21 +239,19 @@ public void Clone_ReturnsMutable()
         public void Enumerator()
         {
             var list = new RepeatedField<string> { "first", "second" };
-            using (var enumerator = list.GetEnumerator())
-            {
-                Assert.Throws<InvalidOperationException>(() => enumerator.Current.GetHashCode());
-                Assert.IsTrue(enumerator.MoveNext());
-                Assert.AreEqual("first", enumerator.Current);
-                Assert.IsTrue(enumerator.MoveNext());
-                Assert.AreEqual("second", enumerator.Current);
-                Assert.IsFalse(enumerator.MoveNext());
-                Assert.Throws<InvalidOperationException>(() => enumerator.Current.GetHashCode());
-                Assert.IsFalse(enumerator.MoveNext());
-                enumerator.Reset();
-                Assert.Throws<InvalidOperationException>(() => enumerator.Current.GetHashCode());
-                Assert.IsTrue(enumerator.MoveNext());
-                Assert.AreEqual("first", enumerator.Current);
-            }
+            var enumerator = list.GetEnumerator();
```

Comment:
Ah, having read http://stackoverflow.com/questions/4642665 I see the issue.
I'll have a think about it. Another option could be to use a try/finally block manually, which might fix it.

Will certainly look at whether the mutable struct provides any benefit any more within the runtime - it still would for clients, in normal situations. (The capture here isn't particularly "normal"...)